src/main/java/redis/clients/jedis/BinaryJedisCluster.java

```diff
@@ -2270,4 +2270,14 @@ public Object execute(Jedis connection){
       }
     }.runBinary(sampleKey);
   }
+  
+
+  public Pipeline pipelined(final byte[] sampleKey) {
+    return new JedisClusterCommand<Pipeline>(connectionHandler, maxAttempts) {
+      @Override
+      public Pipeline execute(Jedis connection){
+        return connection.pipelined();
```

Comment:
Returned `Pipeline` object holds the `Client` object from `Jedis` object (connection). But that `Jedis` object gets returned to pool by JedisClusterCommand.releaseConnection(). This can have severe consequences in (heavy) multi-threaded scenario.

Reply:
@sazzad16 good cache! my bad! I'll rework this solution